worker/src/enclave/worker_api_direct_server.rs

```diff
@@ -210,66 +209,101 @@ pub fn handle_direct_invocation_request(req: DirectWsServerRequest) -> Result<()
-            let old_result: Vec<u8> = event.result.clone();
-            let mut result = RpcReturnValue::decode(&mut old_result.as_slice()).unwrap();
-
-            match status_update {
-                TrustedOperationStatus::Invalid
-                | TrustedOperationStatus::InSidechainBlock(_)
-                | TrustedOperationStatus::Finalized
-                | TrustedOperationStatus::Usurped => {
-                    // Stop watching
-                    result.do_watch = false;
-                    continue_watching = false;
+    // Not removing this function, as we may need this function later. So commented it out
+    /*    let mut status_update_slice =
+            slice::from_raw_parts(status_update_encoded, status_size as usize);
+        let status_update = TrustedOperationStatus::decode(&mut status_update_slice).unwrap();
+        let mut hash_slice = slice::from_raw_parts(hash_encoded, hash_size as usize);
+        if let Ok(hash) = Hash::decode(&mut hash_slice) {
+            // Aquire watched list lock
+            let mutex = load_watched_list().unwrap();
+            let mut watch_list = mutex.lock().unwrap();
+            let mut continue_watching = true;
+            if let Some(client_event) = watch_list.get_mut(&hash) {
+                let mut event = &mut client_event.response;
+                // Aquire result of old RpcResponse
+                let old_result: Vec<u8> = event.result.clone();
+                let mut result = RpcReturnValue::decode(&mut old_result.as_slice()).unwrap();
+
+                match status_update {
+                    TrustedOperationStatus::Invalid
+                    | TrustedOperationStatus::InSidechainBlock(_)
+                    | TrustedOperationStatus::Finalized
+                    | TrustedOperationStatus::Usurped => {
+                        // Stop watching
+                        result.do_watch = false;
+                        continue_watching = false;
+                    }
+                    _ => {}
+                };
+                // update response
+                result.status = DirectRequestStatus::TrustedOperationStatus(status_update);
+                event.result = result.encode();
+                client_event
+                    .client
+                    .send(serde_json::to_string(&event).unwrap())
+                    .unwrap();
+
+                if !continue_watching {
+                    client_event.client.close(CloseCode::Normal).unwrap();
                 }
-                _ => {}
-            };
-            // update response
-            result.status = DirectRequestStatus::TrustedOperationStatus(status_update);
-            event.result = result.encode();
-            client_event
+            } else {
+                continue_watching = false;
+            }
+            if !continue_watching {
+                watch_list.remove(&hash);
+            }
+        }
+    */
+    sgx_status_t::SGX_SUCCESS
```

Comment:
this is dangerous habit if this kind of code makes it to production by merging accidents. How long will it take to fix it properly?

Reply:
for now.. I removed body